crates/burn-import/src/burn/node/eye_like.rs

```diff
@@ -0,0 +1,129 @@
+use super::{Node, NodeCodegen};
+use crate::burn::{Scope, TensorType, Type};
+use burn::record::PrecisionSettings;
+use proc_macro2::TokenStream;
+use quote::quote;
+
+#[derive(Debug, Clone, new)]
+pub struct EyeLikeNode {
+    pub input: TensorType,
+    pub output: TensorType,
+}
+
+impl<PS: PrecisionSettings> NodeCodegen<PS> for EyeLikeNode {
+    fn input_types(&self) -> Vec<Type> {
+        vec![Type::Tensor(self.input.clone())]
+    }
+
+    fn output_types(&self) -> Vec<Type> {
+        vec![Type::Tensor(self.output.clone())]
+    }
+
+    fn forward(&self, scope: &mut Scope, node_position: usize) -> TokenStream {
+        let input = scope.tensor_use_owned(&self.input, node_position);
+        let output = &self.output.name;
+
+        quote! {
+            let #output = {
+                let shape = #input.shape();
+                let device = #input.device();
+
+                // EyeLike creates an identity matrix with the same shape as input
+                assert!(shape.dims.len() == 2, "EyeLike operation requires 2D input tensor");
+
+                let rows = shape.dims[0];
+                let cols = shape.dims[1];
+
+                if rows == cols {
+                    // Square matrix - use Tensor::eye directly
+                    Tensor::eye(rows, &device)
+                } else {
+                    // Rectangular matrix - create zeros and place eye in top-left
+                    let min_dim = core::cmp::min(rows, cols);
+                    let eye_matrix = Tensor::eye(min_dim, &device);
+                    let result = #input.zeros_like();
+
+                    // Use slice assignment to place the identity matrix in the top-left corner
+                    result.slice_assign([0..min_dim, 0..min_dim], eye_matrix)
```

Comment:
Probably you can use https://burn.dev/docs/burn/tensor/struct.Tensor.html#method.diag_mask instead and convert bools to `int`